src/inject.js

```diff
@@ -7,35 +7,27 @@ import { MobXProviderContext } from "./Provider"
  * Store Injection
  */
 function createStoreInjector(grabStoresFn, component, injectNames, makeReactive) {
-    let displayName = getInjectName(component, injectNames)
-
-    class Injector extends Component {
-        static contextType = MobXProviderContext
-
-        render() {
-            const { forwardRef, ...props } = this.props
+    // Support forward refs
+    let Injector = React.forwardRef((props, ref) => {
+        const newProps = { ...props }
+        const context = React.useContext(MobXProviderContext)
+        Object.assign(newProps, grabStoresFn(context || {}, newProps) || {})
```

Comment:
Slightly more concise and done within a single operation. You can drop a line 12.
```suggestion
        const nextProps = Object.assign({}, props, grabStoresFn(context || {}, props) || {})
```

Reply:
We cannot do it because `grabStoresByName` modifies `props` passed as the second argument https://github.com/mobxjs/mobx-react/blob/master/src/inject.js#L66.

Reply:
I see. Ok, let's leave as you did it. Not worth an extra hassle.